### ISSUE TITLE
when use multi_gpu_model, the loss and accuracy became low at the same time since from the second epoch

### ISSUE THREAD

**Issue by dodgaga** [keras-team/keras#13557]
When I don't use **multi_gpu_model** ,everything is ok, but, with the **multi_gpu_model**, training and validation loss as below:
1055/1061 [============================>.] - ETA: 5s - loss: 3.5930 - acc: 0.0740
1056/1061 [============================>.] - ETA: 4s - loss: 3.5926 - acc: 0.0743
1057/1061 [============================>.] - ETA: 3s - loss: 3.5921 - acc: 0.0743
1058/1061 [============================>.] - ETA: 2s - loss: 3.5919 - acc: 0.0744
1059/1061 [============================>.] - ETA: 1s - loss: 3.5917 - acc: 0.0745
1060/1061 [============================>.] - ETA: 0s - loss: 3.5915 - acc: 0.0746
1061/1061 [==============================] - 919s 867ms/step - loss: 3.5911 - acc: 0.0748 - **val_loss: 1.1921e-07 - val_acc: 0.0207**
save weights file ./model_snapshots_multi/weights_000_0.0207.h5
Epoch 2/60

1/1061 [..............................] - ETA: 12:46 - loss: 3.2449 - acc: 0.1875
2/1061 [..............................] - ETA: 12:55 - loss: 3.3741 - acc: 0.1250
3/1061 [..............................] - ETA: 13:06 - loss: 3.2832 - acc: 0.1667
4/1061 [..............................] - ETA: 13:00 - loss: 3.2520 - acc: 0.1719
5/1061 [..............................] - ETA: 13:00 - loss: 3.2117 - acc: 0.1875
6/1061 [..............................] - ETA: 12:26 - loss: 3.3068 - acc: 0.1562
7/1061 [..............................] - ETA: 12:30 - loss: 2.8344 - acc: 0.1339
8/1061 [..............................] - ETA: 12:33 - loss: 2.4801 - acc: 0.1172
9/1061 [..............................] - ETA: 12:35 - loss: 2.2046 - acc: 0.1181
10/1061 [..............................] - ETA: 12:33 - loss: 1.9841 - acc: 0.1062
11/1061 [..............................] - ETA: 12:34 - loss: 1.8037 - acc: 0.0966
12/1061 [..............................] - ETA: 12:36 - loss: 1.6534 - acc: 0.0938
13/1061 [..............................] - ETA: 12:38 - loss: 1.5262 - acc: 0.0865
14/1061 [..............................] - ETA: 12:39 - loss: 1.4172 - acc: 0.0804
15/1061 [..............................] - ETA: 12:38 - loss: 1.3227 - acc: 0.0750
16/1061 [..............................] - ETA: 12:38 - loss: 1.2401 - acc: 0.0703
17/1061 [..............................] - ETA: 12:38 - loss: 1.1671 - acc: 0.0662
18/1061 [..............................] - ETA: 12:38 - loss: 1.1023 - acc: 0.0660
19/1061 [..............................] - **ETA: 12:38 - loss: 1.0443 - acc: 0.0625**
20/1061 [..............................] - **ETA: 12:38 - loss: 0.9921 - acc: 0.0625**
21/1061 [..............................] - **ETA: 12:38 - loss: 0.9448 - acc: 0.0595**

in the first epoch, the training loss and accuracy seems normally. However, at the end of the first epoch, the validation loss and accuracy is weird.  Then, at the second epoch, the training loss and accuracy is weird as well. 

**### my code is here:**

def model_fn(FLAGS, objective, optimizer, metrics):
    model = EfficientNetB5(weights=None,
                           include_top=False,
                           input_shape=(FLAGS.input_size, FLAGS.input_size, 3), 
                           classes=FLAGS.num_classes,
                           pooling=max)
    model.load_weights('/nfs/project/wudao/garbage_cls/garbage_classify-master/models/efficientnet-b5_notop.h5')
    for i, layer in enumerate(model.layers):
        if "batch_normalization" in layer.name:
            model.layers[i] = GroupNormalization(groups=32, axis=-1, epsilon=0.00001)
    x = model.output
    x = GlobalAveragePooling2D()(x)
    x = Dropout(0.4)(x)
    predictions = Dense(FLAGS.num_classes, activation='softmax')(x)  # activation="linear",activation='softmax'
    model = Model(input=model.input, output=predictions)   
    parallel_model = multi_gpu_model(model, 4)  # 修改成自身需要的GPU数量，4代表用4个GPU同时加载程序
    parallel_model.compile(loss=objective, optimizer=optimizer, metrics=metrics)    
    return model, parallel_model`

class LossHistory(Callback):
    def __init__(self, FLAGS, model):
        super(LossHistory, self).__init__( )
        self.FLAGS = FLAGS
        self.model_to_save = model

    def on_train_begin(self, logs={}):
        self.losses = []
        self.val_losses = []

    def on_epoch_end(self, epoch, logs={}):
        self.losses.append(logs.get('loss'))
        self.val_losses.append(logs.get('val_loss'))
        save_path = os.path.join(self.FLAGS.train_local, 'weights_%03d_%.4f.h5' % (epoch, logs.get('val_acc')))
        #self.model.save_weights(save_path)
        self.model_to_save.save_weights(save_path)
        print('save weights file', save_path)
        if self.FLAGS.keep_weights_file_num > -1:
            weights_files = glob(os.path.join(self.FLAGS.train_local, '*.h5'))
            if len(weights_files) >= self.FLAGS.keep_weights_file_num:
                weights_files.sort(key=lambda file_name: os.stat(file_name).st_ctime, reverse=True)` 

##main  training
    ......
    train_sequence, validation_sequence = data_flow(FLAGS.data_local, FLAGS.batch_size,
                                                    FLAGS.num_classes, FLAGS.input_size)
    # optimizer = adam(lr=FLAGS.learning_rate, clipnorm=0.001)
    optimizer = Nadam(lr=FLAGS.learning_rate, beta_1=0.9, beta_2=0.999, epsilon=1e-08, schedule_decay=0.004)
    # optimizer = SGD(lr=FLAGS.learning_rate, momentum=0.9)
    objective = 'categorical_crossentropy'
    metrics = ['accuracy']
    model_orig, model = model_fn(FLAGS, objective, optimizer, metrics)
    ......
    history = LossHistory(FLAGS,model_orig)
    model.fit_generator(
        train_sequence,
        steps_per_epoch=len(train_sequence),
        epochs=FLAGS.max_epochs,
        verbose=1,
        callbacks=[history, tensorBoard, warm_up_lr],
        validation_data=validation_sequence,
        max_queue_size=10,
        workers=int(multiprocessing.cpu_count() * 0.7),
        use_multiprocessing=True,
        shuffle=True
    )

Thanks in advance! Hope someone can help me!!!
